src/Kiota.Builder/Writers/Python/PythonConventionService.cs

```diff
@@ -0,0 +1,171 @@
+        var collectionSuffix = code.CollectionKind == CodeTypeCollectionKind.None && includeCollectionInformation ? string.Empty : "]";
+        if(code is CodeUnionType currentUnion && currentUnion.Types.Any())
+            return currentUnion.Types.Select(x => GetTypeString(x, targetElement, true, writer)).Aggregate((x, y) => $"Union[{x}, {y.ToFirstCharacterLowerCase()}]");
+        else if(code is CodeType currentType) {
+            var typeName = GetTypeAlias(currentType, targetElement) ?? TranslateType(currentType);
+            if (TypeExistInSameClassAsTarget(code, targetElement))
+                typeName = targetElement.Parent.Name.ToFirstCharacterUpperCase();
+            if (code.ActionOf)
+                return WriteInlineDeclaration(currentType, targetElement, writer);
+            else
+                return $"{collectionPrefix}{typeName}{collectionSuffix}";
+        }
+        else throw new InvalidOperationException($"type of type {code.GetType()} is unknown");
+    }
+    #pragma warning restore CA1822 // Method should be static
+    internal static string RemoveInvalidDescriptionCharacters(string originalDescription) => originalDescription?.Replace("\\", "/");
+    public override string TranslateType(CodeType type) {
+        if (type.IsExternal)
+            return TranslateExternalType(type);     
+        else
+            return TranslateInternalType(type);
+    }
+    private static string TranslateExternalType(CodeType type) {
+        return type.Name switch  {
+                "String" or "string" => "str",
+                "integer" or "int32" or "int64" or "byte" or "sbyte" => "int",
+                "decimal" or "double" => "float",
+                "Binary" or "binary" => "bytes",
+                "void" => "None",
+                "DateTimeOffset" => "datetime",
+                "boolean" => "bool",
+                "Object" or "object" or "float" or "bytes" or "datetime" or "timespan" => type.Name,
+                _ => type.Name.ToFirstCharacterUpperCase().Replace("IParseNode", "ParseNode") ?? "object",
+            };
+    }
+    private static string TranslateInternalType(CodeType type) {
+        if (type.Name.Contains("RequestConfiguration"))
+            return type.TypeDefinition?.Name.ToFirstCharacterUpperCase();
+        else if (type.Name.Contains("QueryParameters"))
+            return type.Name;
+        else
+            return type.Name switch  {
+                "String" or "string" => "str",
+                "integer" or "int32" or "int64" or "byte" or "sbyte" => "int",
+                "decimal" or "double" => "float",
+                "Binary" or "binary" => "bytes",
+                "void" => "None",
+                "DateTimeOffset" => "datetime",
+                "boolean" => "bool",
+                "Object" or "object" or "float" or "bytes" or "datetime" or "timespan" => type.Name,
+                _ => $"{type.Name.ToSnakeCase()}.{type.Name.ToFirstCharacterUpperCase()}" ?? "object",
```

Comment:
assuming the replacement is done in the refiner, this block can be replaced by calling the other method

Reply:
This will not work as the default return type `$"{type.Name.ToSnakeCase()}.{type.Name.ToFirstCharacterUpperCase()}"` is different from the other method and works only for internal imports.